packages/api/src/external/fhir/shared/hydrate-bundle.ts

```diff
@@ -0,0 +1,48 @@
+import { Extension, Organization, Patient } from "@medplum/fhirtypes";
+import { isValidUuid } from "@metriport/shared";
+import { metriportDataSourceExtension } from "@metriport/core/external/fhir/shared/extensions/metriport";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { Bundle as ValidBundle } from "../../../routes/medical/schemas/fhir";
+
+/**
+ * Adds the Metriport and Document extensions to all the provided resources, ensures that all resources have UUIDs for IDs,
+ * and adds the Patient and Organization resources to the Bundle
+ */
+export function hydrateBundle(
+  bundle: ValidBundle,
+  patient: Patient,
+  org: Organization,
+  fhirBundleDestinationKey: string
+): ValidBundle {
+  const docExtension: Extension = {
+    url: "https://public.metriport.com/fhir/StructureDefinition/doc-id-extension.json",
+    valueString: fhirBundleDestinationKey,
+  };
+
+  const bundleWithExtensions = ensureUuidsAndAddExtensions(bundle, docExtension);
+  bundleWithExtensions.entry?.push({ resource: patient });
+  bundleWithExtensions.entry?.push({ resource: org });
+
+  return bundleWithExtensions;
+}
+
+type ReplacementIdPair = { old: string; new: string };
+
+function ensureUuidsAndAddExtensions(bundle: ValidBundle, docExtension: Extension): ValidBundle {
```

Comment:
This is not the clearest function I've written.. 
It's currently doing 2 things: 
- Adding extensions to each resource
- Taking note of invalid IDs, and then replaces them

I was thinking of splitting this up into 2 separate functions, but then that is going to double the number of loops we do thru each bundle.... Not sure which of the lesser evils to choose. Unless, there's a good suggestion that prevents ugliness while keeping the functionality

Reply:
maybe
`validateUuidsAndAddExtensions` I think you should keep it